### ISSUE TITLE
[question] Are Vue components being server side rendered?

### ISSUE THREAD

**Issue by OwenVey** [vitebook/vitebook#41]
Im noticing the few components in my Vue library that are using _other_ third party libraries are failing to render in dev mode due to errors like:
`TypeError: Cannot read properties of undefined (reading 'switchRef')`
`TypeError: Cannot set properties of null`
as well as seeing warnings like
`provide() can only be used inside setup()`.

The two libraries in question here are `@headlessui/vue` and `vuedraggable`. I have all of this working in a Vitepress project but was hoping to switch to Vitebook as I love the flexibility and focus on component libraries.

Basically wondering if there are some extra steps I need to take to view these components when in development. The components I'm developing are intended to be used client side and was hoping to keep them built that way.

Love the project, keep up the great work!

**Comment by mihar-22** [comment]
These same libraries work in Vitepress? That's suprising since that also runs server-side rendering to statically generate site contents (SSG). Oh maybe they've added an SPA option? At the moment Vitebook is an SSG and I'd need to time to introduce an SPA option. In the meantime there's two options:

Option 1. Libraries _should_ be SSR-friendly and these issues can be brought up with the offending libraries, seems like simple fixes.

Option 2. Wrap the components or entire stories to only run client-side.

```
<ClientOnly>
    <!-- ... -->
</ClientOnly>
```

Copy the [`<ClientOnly>`](https://github.com/vuejs/vitepress/blob/main/src/client/app/components/ClientOnly.ts) component from Vitepress. We can make this standard and add it to the `@vitebook/vue` package as well.

**Comment by OwenVey** [comment]
Using Vitepress's `<ClientOnly />` component did the trick. I think this would be something nice to include out of the box. Thank again for the help!
